lib/widgets/read_receipts.dart

```diff
@@ -0,0 +1,208 @@
+      setState(() {});
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO could pull out this layout/appearance code,
+    //   focusing this widget only on state management
+
+    return SizedBox(
+      height: 500, // TODO(design) tune
+      child: Column(
+        children: [
+          _ReadReceiptsHeader(receiptCount: userIds.length, status: status),
+          Expanded(child: _ReadReceiptsUserList(userIds: userIds, status: status)),
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 16),
+            child: const BottomSheetDismissButton(style: BottomSheetDismissButtonStyle.close))
+        ]));
+  }
+}
+
+enum FetchStatus { loading, success, error }
+
+class _ReadReceiptsHeader extends StatelessWidget {
+  const _ReadReceiptsHeader({required this.receiptCount, required this.status});
+
+  final int receiptCount;
+  final FetchStatus status;
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    WidgetBuilderFromTextStyle? headerSubtitleBuilder;
+    if (status == FetchStatus.success && receiptCount > 0) {
+      headerSubtitleBuilder = (TextStyle style) => TextWithLink(
+        onTap: () {
+          PlatformActions.launchUrl(context, PerAccountStoreWidget.of(context)
+            .tryResolveUrl(ReadReceipts._helpCenterReference)!);
+        },
+        style: style,
+        markup: zulipLocalizations.actionSheetReadReceiptsReadCount(receiptCount));
+    }
+
+    return Padding(
+      padding: const EdgeInsets.only(top: 8),
+      child: BottomSheetHeader(
+        title: zulipLocalizations.actionSheetReadReceipts,
+        buildSubtitle: headerSubtitleBuilder));
+  }
+}
+
+class _ReadReceiptsUserList extends StatelessWidget {
+  const _ReadReceiptsUserList({required this.userIds, required this.status});
+
+  final List<int> userIds;
+  final FetchStatus status;
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = ZulipLocalizations.of(context);
+
+    return switch(status) {
+      FetchStatus.loading => BottomSheetEmptyContentPlaceholder(loading: true),
+      FetchStatus.error   => BottomSheetEmptyContentPlaceholder(
+        message: localizations.actionSheetReadReceiptsErrorReadCount),
+      FetchStatus.success => userIds.isEmpty
+        ? BottomSheetEmptyContentPlaceholder(
+            message: localizations.actionSheetReadReceiptsZeroReadCount)
+        : InsetShadowBox(
+            top: 8, bottom: 8,
+            color: DesignVariables.of(context).bgContextMenu,
+            child: ListView.builder(
+              padding: EdgeInsets.symmetric(vertical: 8),
+              itemCount: userIds.length,
+              itemBuilder: (context, index) =>
+                ReadReceiptsUserItem(userId: userIds[index])))
+    };
+  }
+}
+
+
+// TODO: deduplicate the code with [ViewReactionsUserItem]
+@visibleForTesting
+class ReadReceiptsUserItem extends StatelessWidget {
```

Comment:
This comment should get a twin at the other end, pointing back to this one.

That way when editing the other one, we might notice and look at this one to edit it the same way (or to go ahead and do this deduplication).